[CELEBORN-1458][DOC] Introduce decommissioning document (#2554)
apache/celeborn: docs/workerdecommission.md

accept new shuffle slot requests or new data. This ensures that no new tasks are assigned to
the workers that are set to be decommissioned.

- Existing Data Handling: The worker nodes will continue to handle their existing shuffle partitions
until all the partitions have expired. This mechanism ensures that current jobs running on these nodes
can complete their data shuffle operations without interruption.

- Worker Exit: After all existing shuffle partitions on the worker nodes have expired,
the workers will gracefully exit. This ensures that the node is safely removed from the cluster
without causing data loss or job failures.

This decommissioning process is essential for maintaining cluster health and efficiency,
as it allows for the smooth removal of unhealthy nodes and enables dynamic resizing of the cluster
to meet varying workload demands.

## Decommission Configuration

| Key                                               | Value |
|---------------------------------------------------|-------| 
| celeborn.worker.decommission.forceExitTimeout     | 6h    |
| celeborn.worker.decommission.checkInterval        | 30s   |

> ## Initiate Decommissioning

Reviewer: ```suggestion
## Perform Decommissioning
```

> An admin sends a decommissioning request to the RESTful API endpoint:

Reviewer: ```suggestion
Administrators perform decommissioning operation via the RESTful API endpoint:
```

> ```text

Reviewer: ```suggestion
```shell
```

Author: Done

curl --request POST --url 'ip:port/exit' --data '{"type":"Decommission"}'
```

> RESTful API can refer to [Restful API](../monitoring/#worker_1)

Reviewer: ```suggestion
Details of decommissioning interface refer to [Restful API](../monitoring/#worker_1).
```

Author: Done

> ## Monitor Worker Status

Reviewer: ```suggestion
## Decommission Monitoring
```

Administrators can monitor the status of the workers to ensure they are gracefully exiting
after all tasks are complete.

> Administrators can monitor the status of the workers under decommission through Restful API [ip:port/isDecommissioning](../monitoring/#worker_1)

Reviewer: Please add the monitor way from Celeborn Master side.

Author: Done
